lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -239,7 +240,7 @@ def discover_resources
 
         split_templates(filename) do |r_def|
           r = KubernetesResource.build(namespace: @namespace, context: @context, logger: @logger,
-                                       definition: r_def, extra_statsd_tags: statsd_tags)
+                                       definition: r_def, statsd_tags: statsd_tags)
```

Comment:
We might as well only pass in the `@namespace_tags` if we have them, this avoids any potential overwrite in the `KubernetesResource` class.

Reply:
Not sure I follow, how would you change it?

Reply:
Addressed IRL, thanks!